supportingami/sami-website
# Feat/static site export

#### What does this PR do?
Make the site static-site-generator (SSG) compatible, so that a nextJS can be used in _export_ mode, generating a single static build that includes all required html/js and assets for the site and can be hosted on any webserver.
https://nextjs.org/docs/pages/building-your-application/deploying/static-exports

**Main Changes**
- Refactor NextJS code to use static props
- Update blog posts and projects endpoints to support slugs
- Update blog post and project single entry to retrieve from slug, and generate full list of static pages during build
- Create additional next.config.js config to use in SSG export
- Add custom image optimisation scripts to work with SSG export (nextJS image optimiser not compatible)
- Add custom scripts to build SSG deployment
- Add scripts and configuration to deploy static site to Vercel hosting
- Update deployment documentation

**Additional Changes**
- Fix footer UI issue forcing small devices to show large page margins
- Core dependency updates
- Improve scripts to work in CI environment (GH actions)

#### How should this be manually tested?

- `yarn build`
- Interactive cli should help to build export version, preview locally, and (optionally) deploy to vercel

The current site has already been deployed to https://next.samicharity.co.uk/

#### What are the relevant github issues?

#### Screenshots
